File: xmp/ZeroconfEmitter.js
#!/usr/bin/env node
var gepard = require ( "gepard" ) ;
var name   = gepard.getProperty ( "name", "ALARM" ) ;
var client = gepard.getClient ( 'test-gepard', function acceptService ( service )
{
	if ( service.getTopics().indexOf ( name ) < 0 )
	{
		return ;
	}
	client.emit ( name,
	{
	  write: function() // The event is sent -> end connection and exit
	  {
	    client.end() ;
	  }
	});
	return true ;
} ) ;

File: python/xmp/ZeroconfEmitter.py
#!/usr/bin/env python

import os, sys, inspect
sys.path.insert ( 0, os.path.dirname(os.path.abspath(__file__) ) + "/../" )
import gepard

gepard.util.exitWithSIGINT()

def acceptService ( client, service ):
	print ( service )
	if service.isLocalHost():
		client.emit ( "ALARM" )
		client.close()
		return True
	return False

client = gepard.Client.getInstance ( 'test-gepard', acceptService )

